### PR TITLE
[SYCL][libdevice] Add SIMD emulation APIs for imf libdevice

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

**Comment by v-klochkov** [comment]
Is there some testing plan for this fix (LIT/unit tests)?

File: libdevice/imf_utils/simd_emulate.cpp [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+      res_buf[idx] = 0x7FFF;
+    else {
+      int16_t s_tmp = __bit_cast<int16_t>(tmp);
+      res_buf[idx] = __bit_cast<uint16_t>(__abs(s_tmp));
+    }
+  }
+  return __assemble_integral_value<unsigned int, uint16_t, 2>(res_buf);
+}
+
+DEVICE_EXTERN_C_INLINE
```
[the remark this answers is not in the document]
Comment (jinge90) [reply libdevice/imf_utils/simd_emulate.cpp:56]: Hi, @v-klochkov 
Thanks for reminder, I will add comments later.

**Comment by jinge90** [comment]
> Is there some testing plan for this fix (LIT/unit tests)?

I am also working on some tests for those imf simd emulate functions and will create PR in intel/llvm-test-suite.

Thanks very much.

**Comment by jinge90** [comment]
Hi, @xtian-github 
This PR includes all SIMD emulate APIs for imf libdevice, could you take a look?

Thanks very much.

File: libdevice/imf_utils/simd_emulate.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jinge90) [reply libdevice/imf_utils/simd_emulate.cpp:382]: Hi, @v-klochkov 
I added comments for all functions, could you help review again?
Thanks very much.

**Comment by jinge90** [comment]
> > Is there some testing plan for this fix (LIT/unit tests)?
> 
> I am also working on some tests for those imf simd emulate functions and will create PR in intel/llvm-test-suite.
> 
> Thanks very much.

Hi, @v-klochkov 
https://github.com/intel/llvm-test-suite/pull/1074 is the test for all functions added, local verification has passed. When this PR is approved and merged, we will merge the test.

Thanks very much.

**Review by xtian-github — DISMISSED** [review]
LGTM

**Review by v-klochkov — DISMISSED** [review]

**Comment by jinge90** [comment]
Hi, @intel/dpcpp-tools-reviewers 
Could you review this PR, we added some new SIMD emulation APIs in libdevice.

Thanks very much.

**Comment by jinge90** [comment]
Hi, @pvchupin 
Could you take a look this PR?
Thanks very much.

**Comment by jinge90** [comment]
Hi, @pvchupin 
Kind ping~

**Comment by jinge90** [comment]
Hi, @intel/dpcpp-tools-reviewers 
Could you take a look of this PR?
Thanks very much.

**Comment by jinge90** [comment]
Hi, @v-klochkov 
Could you help check if this PR can be merged now?
Thanks very much.

**Comment by v-klochkov** [comment]
> Hi, @v-klochkov Could you help check if this PR can be merged now? Thanks very much.

This PR requires approval from @intel/dpcpp-tools-reviewers. @asudarsa - can you review this fix please?

**Comment by pvchupin** [comment]
ping @asudarsa

**Review by asudarsa — APPROVED** [review]
This change looks good to me. Thanks.

**Review by xtian-github — APPROVED** [review]
LGTM

**Comment by pvchupin** [comment]
@jinge90, can you add PR description please?

**Pull request merged**